diffmpm/shapefn.py

```diff
@@ -1,16 +1,108 @@
 import jax.numpy as jnp
+from jax import jacobian, vmap, lax
 
-def Linear1DShapefn(xi):
-    """
-    Linear 1D shape function in natural coordinates
+
+class ShapeFn:
     """
-    shapefn = jnp.array([0.5 * (1 - xi), 0.5 * (1 + xi)])
-    return shapefn
+    Define the shape function of an element.
 
+    Based on the dimension, the shape function returns the required
+    mapped nodal values.
 
-def Linear1DGradsf(xi):
-    """
-    Linear 1D gradient of shape function in natural coordinates.
+    Attributes
+    ----------
+    dim : int
+        Dimension of the shape function.
     """
-    gradsf = jnp.array([-0.5, 0.5])
-    return gradsf
\ No newline at end of file
+
+    def __init__(self, dim=1):
+        """
+        Construct Shape functions for the given dimension.
+
+        Arguments
+        ---------
+        dim : int
+            Dimension of the shape function to be used.
+        """
+        self.dim = dim
+        return
+
+    def shapefn(self, x):
```

Comment:
Could you please use `xi` instead of `x`, since that's the variable we use to define natural coordinates.

Reply:
Okay, got it.